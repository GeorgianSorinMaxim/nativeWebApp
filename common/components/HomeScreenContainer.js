// @flow

import React from 'react';
import { connect } from 'react-redux';

import HomeScreen from './HomeScreen';

type Props = {
  data: Object,
  loading: boolean
};

export const HomeScreenContainer = ({ data, loading }: Props) => (
  <HomeScreen data={data} loading={loading} />
);

const mapStateToProps = state => ({
  data: state.data.entities,
  loading: state.data.loading
});

const HomeScreenContainerWithData = connect(
  mapStateToProps,
  {}
)(HomeScreenContainer);

export default HomeScreenContainerWithData;
