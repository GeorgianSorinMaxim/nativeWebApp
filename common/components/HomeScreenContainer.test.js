import 'react-native';
import React from 'react';
import render from '../utils/testUtils';

import { HomeScreenContainer } from './HomeScreenContainer';

describe('HomeScreenContainer', () => {
  const defaultProps = {
    data: [],
    loading: false
  };

  it('should render', () => {
    const { tree } = render(<HomeScreenContainer {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });
});
