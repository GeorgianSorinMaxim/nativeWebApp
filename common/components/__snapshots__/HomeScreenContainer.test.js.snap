// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreenContainer should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F3F3F3",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 15,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../common/images/vessel.png",
          }
        }
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
            "marginBottom": 15,
            "marginTop": 15,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      This is the ðŸ“± app running on 
      IOS
    </Text>
  </View>
</View>
`;
